tests/unit/preprocessor/_time/test_time.py

```diff
@@ -200,18 +202,16 @@ def test_clip_timerange_3_years(self):
 
     def test_clip_timerange_no_slice(self):
         """Test fail of clip_timerange."""
-        with self.assertRaises(ValueError) as ctx:
-            clip_timerange(self.cube, '2200/2200')
-        msg = ("Time slice 2200-01-01 to 2201-01-01 is outside"
+        self.cube.coord('time').guess_bounds()
+        msg = ("Time slice 2200-01-01 01:00:00 to 2201-01-01 is outside"
                " cube time bounds 1950-01-16 00:00:00 to 1951-12-07 00:00:00.")
         with self.assertRaises(ValueError) as ctx:
-            clip_timerange(self.cube, '2200/2200')
+            clip_timerange(self.cube, '22000101T010000/2200')
```

Comment:
How come this change in the timerange value is needed?

Reply:
To test these two lines:
https://github.com/ESMValGroup/ESMValCore/blob/29336738cb1f67628e5d3814a83fdc0c1a2579cf/esmvalcore/preprocessor/_time.py#L207-L208

By the way, why is a second added to `end_date` here?
https://github.com/ESMValGroup/ESMValCore/blob/29336738cb1f67628e5d3814a83fdc0c1a2579cf/esmvalcore/preprocessor/_time.py#L128

Reply:
>By the way, why is a second added to end_date here?

It's intentional and if I remember well it needed to be added because otherwise the clipping was not including the last timepoint and I could not reproduce the results of the behaviour of the clipping before `timerange` was introduced.